yt_dlp/extractor/bandcamp.py

```diff
@@ -436,45 +435,47 @@ class BandcampWeeklyIE(BandcampIE):  # XXX: Do not subclass from concrete IE
 
     def _real_extract(self, url):
         show_id = self._match_id(url)
-        webpage = self._download_webpage(url, show_id)
-
-        blob = self._extract_data_attr(webpage, show_id, 'blob')
-
-        show = blob['bcw_data'][show_id]
-
-        formats = []
-        for format_id, format_url in show['audio_stream'].items():
-            if not url_or_none(format_url):
-                continue
-            for known_ext in KNOWN_EXTENSIONS:
-                if known_ext in format_id:
-                    ext = known_ext
-                    break
-            else:
-                ext = None
-            formats.append({
-                'format_id': format_id,
-                'url': format_url,
-                'ext': ext,
-                'vcodec': 'none',
-            })
-
-        title = show.get('audio_title') or 'Bandcamp Weekly'
-        subtitle = show.get('subtitle')
-        if subtitle:
-            title += f' - {subtitle}'
+        show_data = self._download_json(
+            'https://bandcamp.com/api/bcradio_api/1/get_show',
+            show_id,
+            'Downloading Show Data.',
+            data=json.dumps({'id': show_id}).encode(),
+            headers={
+                'Content-Type': 'application/json',
+                'Origin': 'https://bandcamp.com',
+                'Referer': 'https://bandcamp.com/',
+            },
+        )
+        raw_metadata = self._extract_data_attr(
+            self._download_webpage(
+                url,
+                show_id,
+                'Downloading Show Metadata.',
+            ), show_id, 'blob',
+        )
+        metadata = traverse_obj(raw_metadata, (
+            'appData', 'shows', lambda _, mb: int(mb.get('showId')) == int(show_id),
+            any))
+
+        audio_data = show_data['radioShowAudio']
+        thum_id = metadata.get('imageId')
+        thum_url = f'https://f4.bcbits.com/img/000{thum_id}_171.jpg' if thum_id else None
 
         return {
             'id': show_id,
-            'title': title,
-            'description': show.get('desc') or show.get('short_desc'),
-            'duration': float_or_none(show.get('audio_duration')),
+            'title': metadata.get('title'),
+            'thumbnail': thum_url or None,
+            'description': metadata.get('desc') or metadata.get('short_desc'),
+            'duration': float_or_none(show_data.get('duration')),
             'is_live': False,
-            'release_date': unified_strdate(show.get('published_date')),
+            'release_date': unified_strdate(metadata.get('date')),
+            'timestamp': unified_timestamp(metadata.get('date')),
```

Comment:
don't add `timestamp` if you're just duplicating `release_date`

I would change `release_date` to `release_timestamp` and add a fallback to the value given in `audio_data`